time/date.jsonnet

```diff
@@ -34,8 +34,52 @@ local months_seconds(year, month) = std.foldl(
 // days_seconds returns the number of seconds in all days up to day-1.
 local days_seconds(day) = (day - 1) * 24 * 3600;
 
+// is_numeric returns a bool indicating whether the input string is numeric
+local is_numeric(input) = std.length(input) > 0 && std.length(std.prune([
+  if std.codepoint('0') > std.codepoint(char) || std.codepoint(char) > std.codepoint('9') then char
+  for char in std.stringChars(input)
+])) == 0;
+
+// parse_date_or_time parses input which has part_names separated by sep.
+local parse_date_or_time(input, sep, part_names) = (
+  local parts = std.split(input, sep);
+  assert std.length(parts) == std.length(part_names) : 'expected %(expected)d parts separated by %(sep)s in %(format)s formatted input "%(input)s", but got %(got)d' % {
+    expected: std.length(part_names),
+    sep: sep,
+    format: std.join(sep, part_names),
+    input: input,
+    got: std.length(parts),
+  };
+
+  {
+    [part_names[i]]:
+      // Fail with meaningful message if not numeric, otherwise it will be a hell to debug.
+      assert is_numeric(parts[i]) : '%(part_name)%s part "%(part)s" of %(format)s of input "%(input)s" is not numeric' % {
+        part_name: part_names[i],
+        part: parts[i],
+        format: std.join(sep, part_names),
+        input: input,
+      };
+      std.parseInt(parts[i])
+    for i in std.range(0, std.length(parts) - 1)
+  }
+);
+
 {
-  // date_to_unix_timestamp transforms a date into a unix timestamp.
+  // to_unix_timestamp transforms a date into a unix timestamp.
   to_unix_timestamp(year, month, day, hour, minute, second)::
     years_seconds(year) + months_seconds(year, month) + days_seconds(day) + hour * 3600 + minute * 60 + second,
+
+  // parse_rfc3339 parses an RFC3339 timestamp in UTC (ending in 'Z') with date and time separated by 'T', like '2006-01-02T15:04:05Z'
```

Comment:
I appreciate that a full RFC parser is perhaps more work than is worth right now to achieve the goals you have.

I wonder if we should name this `parse_rfc3339_utc` to reflect that it's a partial implementation of the full RFC?

I think it would also be useful to recognize that we are not supporting fractional seconds.

Reply:
What if I state the basic things I support now (no timezones, no sub-second, no lowercase `t` and `z`, and no space separator) and also state in a comment that nobody should rely on the fact that if fails parsing those, so we can extend it in the future?

If someone _needs_ those, they can implement them later by extending the implementation.